sql/src/main/java/io/crate/analyze/relations/RelationSplitter.java

```diff
@@ -152,29 +152,45 @@ private void processOrderBy() {
         Multimap<AnalyzedRelation, Integer> splits = Multimaps.newSetMultimap(
                 new IdentityHashMap<AnalyzedRelation, Collection<Integer>>(specs.size()),
                 INT_SET_SUPPLIER);
+
+        // Detect remaining orderBy
```

Comment:
pls add a comment why this must be done upfront instead on the next for loop (e.g. we must detect before pushing anything down)